pkg/identityserver/entity_access.go

```diff
@@ -75,7 +75,7 @@ func (is *IdentityServer) authInfo(ctx context.Context) (info *ttnpb.AuthInfoRes
 
 	md := rpcmetadata.FromIncomingContext(ctx)
 	if md.AuthType == "" {
-		return &ttnpb.AuthInfoResponse{}, nil
+		return nil, nil
```

Comment:
I think that the `nil` here is not ideal. In general returning `nil` over gRPC is not encouraged.

I think that a user being authenticated means that either:
- `res.UniversalRights` is not empty OR
- `res.AccessMethod` is not `nil`

The empty message (that was used here before) would be compatible with this check (since `AccessMethod` would be `nil`).